app/chaturbate/main.go

```diff
@@ -54,50 +58,71 @@ func main() {
 	go fastStart()
 
 	const SOCK = "/tmp/statbate.sock"
-	os.Remove(SOCK)
+	if err := os.Remove(SOCK); err != nil {
+		logErrorf("socket err: %v", err)
+		return
+	}
 	unixListener, err := net.Listen("unix", SOCK)
 	if err != nil {
-		log.Fatal("Listen (UNIX socket): ", err)
+		logErrorf("socket err: %v", err)
+		return
+	}
+	defer func() {
+		if err = unixListener.Close(); err != nil {
+			logErrorf("socker err: %v", err)
+			return
+		}
+	}()
+
+	if err = os.Chmod(SOCK, os.FileMode((0o777))); err != nil {
+		logErrorf("socket err: %v", err)
+		return
+	}
+	if err = http.Serve(unixListener, nil); err != nil {
+		logErrorf("socket err: %v", err)
+		return
 	}
-	defer unixListener.Close()
-	os.Chmod(SOCK, 0777)
-	log.Fatal(http.Serve(unixListener, nil))
 }
 
 func initMysql() {
 	db, err := sqlx.Connect("mysql", conf.Conn["mysql"])
 	if err != nil {
-		panic(err)
+		logFatalf("database mysql err: %v", err)
 	}
 	Mysql = db
 }
 
 func initClickhouse() {
 	db, err := sqlx.Connect("clickhouse", conf.Conn["click"])
 	if err != nil {
-		panic(err)
+		logFatalf("database clickhouse err: %v", err)
 	}
 	Clickhouse = db
 }
 
 func wJson(s string) {
-	os.WriteFile("/tmp/fastStart.txt", []byte(s), 0644)
+	if err := os.WriteFile("/tmp/fastStart.txt", []byte(s), os.FileMode(0o644)); err != nil {
+		logErrorf("fs err: %v", err)
+	}
 }
 
 func fastStart() {
 	val, err := os.ReadFile("/tmp/fastStart.txt")
 	if err != nil {
-		fmt.Println(err)
+		logErrorf("fs err: %v", err)
 		return
 	}
 	list := make(map[string]*Info)
 	if err := json.Unmarshal(val, &list); err != nil {
-		fmt.Println(err.Error())
+		logErrorf("json err: %v", err)
 		return
 	}
 	for k, v := range list {
 		fmt.Println("fastStart:", k, v.Server, v.Proxy)
-		http.Get("https://statbate.com/cmd/?room=" + k + "&server=" + v.Server + "&proxy=" + v.Proxy)
+		rsp, err := http.Get("https://statbate.com/cmd/?room=" + k + "&server=" + v.Server + "&proxy=" + v.Proxy)
+		if err != nil || rsp.StatusCode >= 400 {
```

Comment:
in case of 401/403/404 we have err == nil , but rsp.StatusCode contains actual http code, normal http responses contains StatusCode < 400